xwiki-pro-macros-ui/src/main/resources/XWiki/Macros/Button.xml

```diff
@@ -322,6 +322,9 @@ $services.localization.render('rendering.macro.button.description')
   #set ($darkenedColors = [])
   #darken($colors, $darkenedColors)
   {{html clean=false}}
+  ## #124: The Button macro deletes followup content when inserted from wysiwyg
+  ## If not in inline mode, wrap inside block in order to fix issue
```

Comment:
```suggestion
  ## Wrap the macro output inside a block element if the context where the macro is inserted is not inline, e.g. if the
  ## macro is inserted between paragraphs rather than between the text of a paragraph. Otherwise the generated HTML is
  ## invalid which breaks the WYSIWYG editing because the browser tries to fix the invalid HTML and messes up the macro
  ## output markers that protect the macro output.
```